deployment/cdk/opensearch-service-migration/lib/migration-assistance-stack.ts

```diff
@@ -41,6 +42,45 @@ export class MigrationAssistanceStack extends Stack {
         }
     }
 
+    validateAndReturnVPCSubnets(vpc: IVpc, brokerNodeCount: number, azCount: number, specifiedSubnetIds?: string[]): string[] {
+        if (specifiedSubnetIds) {
+            if (specifiedSubnetIds.length !== 2 && specifiedSubnetIds.length !== 3) {
+                throw new Error(`MSK requires subnets for 2 or 3 AZs, but have detected ${specifiedSubnetIds.length} subnet ids provided with 'mskSubnetIds'`)
+            }
+            if (brokerNodeCount < 2 || brokerNodeCount % specifiedSubnetIds.length !== 0) {
+                throw new Error(`The MSK broker node count (${brokerNodeCount} nodes inferred) must be a multiple of the number of 
+                    AZs (${specifiedSubnetIds.length} AZs inferred from provided 'mskSubnetIds'). The node count can be set with the 'mskBrokerNodeCount' context option.`)
+            }
+            const selectSubnets = vpc.selectSubnets({
+                subnetFilters: [SubnetFilter.byIds(specifiedSubnetIds)]
+            })
+            return selectSubnets.subnetIds
+        }
+        if (azCount !== 2 && azCount !== 3) {
+            throw new Error(`MSK requires subnets for 2 or 3 AZs, but have detected an AZ count of ${azCount} has been provided with 'mskAZCount'`)
+        }
+        if (brokerNodeCount < 2 || brokerNodeCount % azCount !== 0) {
+            throw new Error(`The MSK broker node count (${brokerNodeCount} nodes inferred) must be a multiple of the number of 
+                AZs (${azCount} AZs inferred from provided 'mskAZCount'). The node count can be set with the 'mskBrokerNodeCount' context option.`)
+        }
+
+        let uniqueAzPrivateSubnets: string[] = []
+        if (vpc.privateSubnets.length > 0) {
+            uniqueAzPrivateSubnets = vpc.selectSubnets({
+                subnetType: SubnetType.PRIVATE_WITH_EGRESS,
+                onePerAz: true
+            }).subnetIds
+        }
+        let desiredSubnets
+        if (uniqueAzPrivateSubnets.length >= azCount) {
+            desiredSubnets = uniqueAzPrivateSubnets.slice(0, azCount)
```

Comment:
Will this list be randomized?  Would every customer get the same AZs back if they were setup in say the same 5 AZs?

Reply:
I'm all for making our code deterministic. The AZ mappings (us-east-1a.. etc.) are randomized by aws account.

Reply:
They should get back the same subnets, but from reading this [issue](https://github.com/aws/aws-cdk/issues/16250), there isn't a firm guarantee. I've added a sort here to give some confidence when the same subnets are used, but if say some subnets are added to the VPC later I believe both cases will cause issues, and maybe this is something we are okay with calling out as a VPC limitation. I'm not aware of a better way to handle this outside of storing state about the subnets ourselves somewhere but that presents other issues.

Reply:
See comment I made about the sort